keras/callbacks.py

```diff
@@ -782,13 +782,24 @@ def is_indexed_slices(grad):
         if self.embeddings_freq:
             embeddings_layer_names = self.embeddings_layer_names
 
-            if not embeddings_layer_names:
-                embeddings_layer_names = [layer.name for layer in self.model.layers
-                                          if type(layer).__name__ == 'Embedding']
+            embedding_layers = set()
+            layers = model.layers[:]
+            while len(layers) > 0:
+                layer = layers.pop(0)
+                if isinstance(layer, __import__(__name__).layers.Embedding):
+                    embedding_layers.add(layer)
+                elif isinstance(layer, __import__(__name__).layers.Wrapper):
```

Comment:
`__import__(__name__)`?

Reply:
Just checking for `Wrapper` will not work in the general case. The only `Wrapper` that could work with an `Embedding` currently is `TimeDistributed` (but even that sounds overly niche). Please remove this statement

Reply:
I used __import__(__name__) because this seemed like the cleanest way to avoid a cyclic import dependency issue and using type(layer).__name__ == ... won't work for subclasses. Is there a better way?

Reply:
Every other way that I could think of to do the same thing involved layers that don't support masking (e.g. Reshape).